src/unique_ptr.rs

```diff
@@ -205,6 +205,19 @@ where
     }
 }
 
+impl<T> PartialEq for UniquePtr<T>
+where
+    T: PartialEq + UniquePtrTarget,
+{
+    fn eq(&self, other: &Self) -> bool {
+        match (self.as_ref(), other.as_ref()) {
+            (None, None) => true,
+            (Some(this), Some(other)) => this == other,
+            _ => false,
+        }
```

Comment:
Simpler implementation:
```suggestion
        self.as_ref() == other.as_ref()
```

Reply:
Updated, thanks for the suggestion